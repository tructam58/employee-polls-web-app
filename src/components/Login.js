// import { InputText } from "primereact/inputtext";
// import { Button } from 'primereact/button';
import { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useNavigate, useLocation } from "react-router-dom";
import { getUsers } from "../slices/authen";
import { currentUser } from "../slices/currentUser";

// import 'primeflex/primeflex.css';

export const Login = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const location = useLocation();
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(false);
  const [success, setSuccess] = useState(false);
  const allUser = useSelector((state) => state.allUser);

  const path = location?.pathname;
  useEffect(() => {
    dispatch(getUsers());
  }, []);
  console.log(path)
  const handleClick = (event) => {
    setPassword(password);
    setUsername(username);
    if(username && password) {
      const user = Object.entries(allUser.value).filter(
        (val) => val[1].id === username && val[1].password === password
      );
      if (user && user.length) {
        dispatch(currentUser(user[0]));
        setError(false);
        setSuccess(true);
       if(path == '/') navigate('/home')
       else navigate(path, { replace: true });
        
      } else {
        setError(true);
        setSuccess(false);
      }
    }
    else {
      setError(true);
      setSuccess(false);
    }
    
  };
  return (
    <div
      className="card"
      style={{
        position: "absolute",
        width: "100%",
        top: "50%",
        textAlign: "center",
        transform: "translateY(-50%)",
      }}
    >
      <div className="text-center">
        <h1>Employee Polls</h1>
        <h1>Login</h1>
        {error && (
          <div className="error" data-testid="fail-id">
            <small>Incorect Username or Password</small>
          </div>
        )}
        {success && (
          <div className="success" data-testid="success-id">
            <small>Success login</small>
          </div>
        )}
      </div>

      <div  style={{ marginBottom: "10px" }}>
        <span>
          <label style={{ marginRight: "8px" }} htmlFor="username">
            Username
          </label>
          <input data-testid="username-input"
            style={{ padding: "8px" }}
            type="text"
            id="username"
            value={username}
            onChange={(e) => setUsername(e.target.value)}
          />
        </span>
      </div>
      <div style={{ marginBottom: "10px" }}>
        <span className="p-float-label">
          <label style={{ marginRight: "8px" }} htmlFor="password">
            Password
          </label>
          <input data-testid="password-input"
            style={{ padding: "8px" }}
            type="text"
            id="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </span>
      </div>
      <div>
        <input type="submit" onClick={handleClick} data-testid='submit' value="Login"/>
      </div>
    </div>
  );
};
