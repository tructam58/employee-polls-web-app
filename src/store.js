import { configureStore } from "@reduxjs/toolkit";
import authen from "./slices/authen";
import user from "./slices/currentUser";
import question from "./slices/question";
 

const store = configureStore({
    reducer: {
        allUser: authen,
        createQuestion: authen,
        logout: user,
        currentUser: user,
        questions: question
    },
});

export default store;