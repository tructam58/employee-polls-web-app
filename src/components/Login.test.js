import { fireEvent, render, screen } from "@testing-library/react"
import { Login } from "./Login";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../store";

import * as React from 'react';
describe('Login page', () => {
  it('will match to snapshot', () => {
    const component = render(<MemoryRouter><Provider store={store}><Login/></Provider></MemoryRouter>);
    expect(component).toMatchSnapshot();
  })

  it('will show a success message if the username and password are correct', () => {
    const component = render(<MemoryRouter><Provider store={store}><Login/></Provider></MemoryRouter>);
    const username = component.getByTestId('username-input');
    const password = component.getByTestId('password-input');
    const submit = component.getByTestId('submit');
    fireEvent.change(username, { target: { value: 'mtsamis'}});
    fireEvent.change(password, { target: { value: 'xyz123'}});
    fireEvent.click(submit);
    expect(component.getByTestId('success-id')).toBeInTheDocument();
  })

  it('will show an error message if the username or password is incorrect', () => {
    const component = render(<MemoryRouter><Provider store={store}><Login/></Provider></MemoryRouter>);
    const submit = component.getByTestId('submit');
    fireEvent.click(submit);
    expect(component.getByTestId('fail-id')).toBeInTheDocument();
  })
})