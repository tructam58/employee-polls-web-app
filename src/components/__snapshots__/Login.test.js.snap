// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page will match to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        style="position: absolute; width: 100%; top: 50%; text-align: center; transform: translateY(-50%);"
      >
        <div
          class="text-center"
        >
          <h1>
            Employee Polls
          </h1>
          <h1>
            Login
          </h1>
        </div>
        <div
          style="margin-bottom: 10px;"
        >
          <span>
            <label
              for="username"
              style="margin-right: 8px;"
            >
              Username
            </label>
            <input
              data-testid="username-input"
              id="username"
              style="padding: 8px;"
              type="text"
              value=""
            />
          </span>
        </div>
        <div
          style="margin-bottom: 10px;"
        >
          <span
            class="p-float-label"
          >
            <label
              for="password"
              style="margin-right: 8px;"
            >
              Password
            </label>
            <input
              data-testid="password-input"
              id="password"
              style="padding: 8px;"
              type="text"
              value=""
            />
          </span>
        </div>
        <div>
          <input
            data-testid="submit"
            type="submit"
            value="Login"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      style="position: absolute; width: 100%; top: 50%; text-align: center; transform: translateY(-50%);"
    >
      <div
        class="text-center"
      >
        <h1>
          Employee Polls
        </h1>
        <h1>
          Login
        </h1>
      </div>
      <div
        style="margin-bottom: 10px;"
      >
        <span>
          <label
            for="username"
            style="margin-right: 8px;"
          >
            Username
          </label>
          <input
            data-testid="username-input"
            id="username"
            style="padding: 8px;"
            type="text"
            value=""
          />
        </span>
      </div>
      <div
        style="margin-bottom: 10px;"
      >
        <span
          class="p-float-label"
        >
          <label
            for="password"
            style="margin-right: 8px;"
          >
            Password
          </label>
          <input
            data-testid="password-input"
            id="password"
            style="padding: 8px;"
            type="text"
            value=""
          />
        </span>
      </div>
      <div>
        <input
          data-testid="submit"
          type="submit"
          value="Login"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
